lib/flex.js

```diff
@@ -181,7 +181,7 @@ const processPreActions = async (context, registry, config) => {
     if (isEventInContext(rule.when, context)) {
       registry.actions.forEach(action => {
         if (action.isEventSupported(`${context.event}.${context.payload.action}`)) {
-          promises.push(action.processBeforeValidate(context))
+          promises.push(action.processBeforeValidate(context, rule, rule.name))
```

Comment:
Why not add rule and rule.name to `context`? That's what it's for so that each function has the context it needs and such that we aren't constantly adding new args.

Reply:
well, context is not individual to each `recipe`. if there are multiple `recipe`, context will have to hold that information separate from the settings

Reply:
Got it. Since you're passing in the rule and name is a property of rule, can't the function obtain `name` from within? Any reason to have the `rule` object and `name` property of rule as two separate args?

Reply:
for `beforeValidate` yes, we can grab the name from `rule` since it doesn't require any specific settings to process beside to log it. in `aftervalidate` we only pass in the relevant settings which is based on whether the validations were a `pass` or `fail`. So we need a `name` field in aftervalidate for sure, i also added in for `beforevalidate` for symmetry.